src/main/java/net/minecraftforge/common/extensions/IForgeLivingEntity.java

```diff
@@ -0,0 +1,69 @@
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.fluids.FluidType;
+
+public interface IForgeLivingEntity extends IForgeEntity
+{
+    default LivingEntity self()
+    {
+        return (LivingEntity) this;
+    }
+
+    @Override
+    default boolean canSwimInFluidType(FluidType type)
+    {
+        if (self().isSensitiveToWater() && type == ForgeMod.WATER_TYPE.get()) return false;
+        else return IForgeEntity.super.canSwimInFluidType(type);
```

Comment:
I think this might read better as:
```java
if (type == ForgeMod.WATER_TYPE.get())
    return !self().isSensitiveToWater();
return IForgeEntity.super.canSwimInFluidType(type);
```

Given I don't think there is any functional difference for water.